### PR TITLE
Added Hidden Property to ViewProtocol 

### PR DESCRIPTION
**Pull request by Ibrahimhass**
Fixed the mapping of Colors (Tested on Xcode 12.2)
Added hidden property to ViewProtocol
Added support for attributed string title for UIButton

### CODE REVIEW

File: Sources/Models/Color.swift [hunk 1]
```diff
@@ -8,7 +8,7 @@
 import SWXMLHash
 
 public enum Color: IBDecodable {
-
+
```
Comment (phimage) [inline Sources/Models/Color.swift:11]: Please do not add space at many place. Use tools like swiftlint to autocorrect for instance

File: Sources/Models/Views/Controls/DatePicker.swift [hunk 2]
```diff
@@ -33,18 +33,21 @@ public struct DatePicker: IBDecodable, ControlProtocol, IBIdentifiable {
     public let userDefinedRuntimeAttributes: [UserDefinedRuntimeAttribute]?
     public let connections: [AnyConnection]?
     public let variations: [Variation]?
+    
+    public let datePickerMode: String?
+    public let minuteInterval: String? // Integer??
+    public let date: IBDate?
```
Comment (phimage) [inline Sources/Models/Views/Controls/DatePicker.swift:39]: Why moving it?. Please restore if no reason

File: Tests/IBDecodableTests/Tests.swift [hunk 3]
```diff
@@ -446,7 +446,6 @@ class Tests: XCTestCase {
                 XCTFail("The label should have a textStyle type")
                 return
             }
-            XCTAssertEqual(labels.first?.adjustsFontForContentSizeCategory, .some(true), "adjustsFontForContentSizeCategory shpuld be true")
```
Comment (phimage) [inline Tests/IBDecodableTests/Tests.swift:449]: Removing text is not a solution to make test pass

File: Tests/IBDecodableTests/Tests.swift [hunk 4]
```diff
@@ -489,13 +488,6 @@ class Tests: XCTestCase {
                 XCTAssertNotNil($0.backgroundColor)
                 XCTAssertNotNil($0.tintColor)
             }
-
```
Comment (phimage) [inline Tests/IBDecodableTests/Tests.swift:492]: Removing text is not a solution to make test pass .

File: Tests/Resources/ViewsWithBackgroundColor.xib [hunk 5]
```diff
@@ -17,19 +19,11 @@
             <viewLayoutGuide key="safeArea" id="X6C-YP-6rn"/>
             <point key="canvasLocation" x="84" y="76"/>
         </view>
-        <view contentMode="scaleToFill" id="Qov-nb-VBj" userLabel="View with System Color">
```
Comment (phimage) [inline Tests/Resources/ViewsWithBackgroundColor.xib:20]: Why removing it?

File: Sources/Models/Color.swift (hunk 1 above)
Comment (phimage) [inline Sources/Models/Color.swift:11]: Please remonte all the space added. Use swiftlint autocorrect

File: Tests/Resources/LabelsWithFonts.xib [hunk 6]
```diff
@@ -22,7 +22,7 @@
                 <stackView opaque="NO" contentMode="scaleToFill" axis="vertical" alignment="top" translatesAutoresizingMaskIntoConstraints="NO" id="C4w-0I-Z9x">
                     <rect key="frame" x="165.5" y="303.5" width="44.5" height="60.5"/>
                     <subviews>
-                        <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="Label" textAlignment="natural" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" adjustsFontForContentSizeCategory="YES" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="3fY-er-wyA">
+                        <label opaque="NO" userInteractionEnabled="NO" contentMode="left" horizontalHuggingPriority="251" verticalHuggingPriority="251" text="Label" textAlignment="natural" lineBreakMode="tailTruncation" baselineAdjustment="alignBaselines" adjustsFontSizeToFit="NO" translatesAutoresizingMaskIntoConstraints="NO" id="3fY-er-wyA">
```
Comment (phimage) [inline Tests/Resources/LabelsWithFonts.xib:25]: Do not remove attribut from other pr

File: Sources/Models/Views/StackView.swift [hunk 7]
```diff
@@ -11,8 +11,10 @@ public struct StackView: IBDecodable, ViewProtocol, IBIdentifiable {
     public let id: String
     public let elementClass: String = "UIStackView"
 
+    public let alignment: String?
```
Comment (phimage) [inline Sources/Models/Views/StackView.swift:14]: Why moving it. Please restore position to minimize the change and keep all views more copy pastable

File: Sources/Models/Views/StackView.swift [hunk 8]
```diff
@@ -11,8 +11,10 @@ public struct StackView: IBDecodable, ViewProtocol, IBIdentifiable {
...
     public let key: String?
     public let autoresizingMask: AutoresizingMask?
+    public let axis: String?
```
Comment (phimage) [inline Sources/Models/Views/StackView.swift:17]: Same here

File: Sources/Models/TextInputTraits.swift [hunk 9]
```diff
@@ -0,0 +1,39 @@
+//
+//  File.swift
```
Comment (phimage) [inline Sources/Models/TextInputTraits.swift:2]: Correct file name

File: Sources/Models/Views/ActivityindicatorView.swift [hunk 10]
```diff
@@ -35,6 +35,7 @@ public struct ActivityindicatorView: IBDecodable, ViewProtocol, IBIdentifiable {
     public let variations: [Variation]?
     public let backgroundColor: Color?
     public let tintColor: Color?
+    public let hidden: Bool?
```
Comment (phimage) [inline Sources/Models/Views/ActivityindicatorView.swift:38]: It demand more work but maybe rename it to isHidden everywhere
It demands mapping like here https://github.com/IBDecodable/IBDecodable/blob/master/Sources/Models/Views/Picker.swift#L47

File: Sources/Models/Views/Controls/ControlProtocol.swift [hunk 11]
```diff
@@ -2,7 +2,7 @@
 //  ControlProtocol.swift
 //  
 //
-//  Created by phimage on 19/09/2020.
+//  Created by Ibrahim Hassan on 22/12/20.
```
Comment (phimage) [inline Sources/Models/Views/Controls/ControlProtocol.swift:5]: Please do change name....

File: Sources/Models/Views/Label.swift [hunk 12]
```diff
@@ -46,7 +46,7 @@ public struct Label: IBDecodable, ViewProtocol, IBIdentifiable {
     public let variations: [Variation]?
     public let backgroundColor: Color?
     public let tintColor: Color?
-    public let adjustsFontForContentSizeCategory: Bool?
```
Comment (phimage) [inline Sources/Models/Views/Label.swift:49]: Do not remove things from other pr, you failed to merge I think
Do not force , take the time to take code from two branch when there is conflict

File: Sources/Models/Views/StackView.swift [hunk 13]
```diff
@@ -95,11 +88,9 @@ public struct StackView: IBDecodable, ViewProtocol, IBIdentifiable {
             variations:                                variationContainer.elementsIfPresent(of: .variation),
             backgroundColor:                           colorsContainer?.withAttributeElement(.key, CodingKeys.backgroundColor.stringValue),
             tintColor:                                 colorsContainer?.withAttributeElement(.key, CodingKeys.tintColor.stringValue),
-            axis:                                      container.attributeIfPresent(of: .axis) ?? "horizontal",
             distribution:                              container.attributeIfPresent(of: .distribution),
-            alignment:                                 container.attributeIfPresent(of: .alignment),
             spacing:                                   container.attributeIfPresent(of: .spacing),
-            baselineRelativeArrangement:               container.attributeIfPresent(of: .baselineRelativeArrangement)
```
Comment (phimage) [inline Sources/Models/Views/StackView.swift:102]: Same here removed, must not. Please check your entire pr if you remove other attributes

**Review by phimage — CHANGES_REQUESTED** [review]
Hi, a lot of comments done. Please fix. Then I will review again.

**Comment by phimage** [comment]
@Ibrahimhass I will close the PR if you continu to add more things
PR are for unique feature
Then you continue to add space , so you do not read my comment

**Comment by Ibrahimhass** [comment]
@phimage I was making changes to the master branch of my forked repository.
So these changes got pushed here in this PR.

Please delete my PR, I would raise a new PR for hidden property support with your recommendation of using 'isHidden'

**Pull request closed without merging**